smallrye-reactive-messaging-aws-sns/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsVerticle.java

```diff
@@ -0,0 +1,148 @@
+import software.amazon.awssdk.services.sns.model.ListSubscriptionsByTopicResponse;
+import software.amazon.awssdk.services.sns.model.SubscribeRequest;
+import software.amazon.awssdk.services.sns.model.SubscribeResponse;
+import software.amazon.awssdk.services.sns.model.Subscription;
+
+/**
+ * Vertx verticle that handles subscription to SNS topic and receive notifications from topic.
+ * 
+ * @author iabughosh
+ * @version 1.0.4
+ *
+ */
+public class SnsVerticle extends AbstractVerticle {
+
+    /** Fields */
+    private final String topicName;
+    private final String endpoint;
+    private final int port;
+    private static final Logger LOG = LoggerFactory.getLogger(SnsVerticle.class);
+    private SnsMessageManager messageManager = new SnsMessageManager();
+    private BlockingQueue<SnsMessage<String>> msgQ = new LinkedBlockingDeque<>();
+
+    /**
+     * Parameterized constructor.
+     * 
+     * @param endpoint Endpoint url.
+     * @param topicName SNS topic name.
+     * @param port listening port for this verticle.
+     */
+    public SnsVerticle(String endpoint, String topicName, int port) {
+        this.topicName = topicName;
+        this.endpoint = endpoint;
+        this.port = port;
+    }
+
+    @Override
+    public void start(Future<Void> startFuture) {
+
+        Router router = Router.router(vertx);
+        router.route()
+                .handler(BodyHandler.create())
+                .method(HttpMethod.POST);
+        router.post(String.format("/sns/%s", topicName)).handler(this::receiveSnsMsg);
+
+        ApacheSdkHttpService apacheSdkHttpService = new ApacheSdkHttpService();
+        SdkHttpClient apacheHttpClient = apacheSdkHttpService.createHttpClientBuilder().build();
+
+        try (SnsClient snsClient = SnsClient.builder().httpClient(apacheHttpClient).build()) {
+            CreateTopicRequest topicCreationRequest = CreateTopicRequest.builder().name(topicName).build();
+            CreateTopicResponse topicCreationResponse = snsClient.createTopic(topicCreationRequest);
+            String topicArn = topicCreationResponse.topicArn();
+            String topicEndpoint = String.format("%s/sns/%s", endpoint, topicName);
+            LOG.info(String.format("Topic ARN is %s, Endpoint is %s", topicArn, topicEndpoint));
+
+            Optional<Subscription> subscription = doesSubscriptionExist(snsClient, topicArn);
+            if (!subscription.isPresent()) {
+                SubscribeResponse response = snsClient.subscribe(SubscribeRequest
+                        .builder()
+                        .topicArn(topicArn)
+                        .endpoint(topicEndpoint)
+                        .protocol("http")
+                        .build());
+                LOG.info(String.format("Topic Subscribtion response %s", response.subscriptionArn()));
+            }
+        }
+
+        vertx.createHttpServer().requestHandler(router).listen(port);
```

Comment:
listen is async.
You must report the success or failure in the `startFuture` object.